fem/fespace.hpp

```diff
@@ -660,76 +781,261 @@ class FiniteElementSpace
+   /// FiniteElementSpace::GetNDofs().
+   ///
+   /// @note The @a dofs array is overwritten and resized to accomodate the
+   /// new values.
    void DofsToVDofs(Array<int> &dofs, int ndofs = -1) const;
 
+   /// @brief Compute the set of @ref vdof "vdofs" corresponding to each entry
+   /// in @a dofs for the given vector index @a vd.
+   ///
+   /// The @a ndofs parameter can be used to indicate the total number of Dofs
+   /// associated with each component of @b vdim. If @a ndofs is -1 (the
+   /// default value), then the number of Dofs is <determined by the
+   /// FiniteElementSpace::GetNDofs().
+   ///
+   /// @note The @a dofs array is overwritten with the new values but its size
+   /// will not be altered.
    void DofsToVDofs(int vd, Array<int> &dofs, int ndofs = -1) const;
 
+   /// @brief Compute a single @ref vdof corresponding to the index @a dof and
+   /// the vector index @a vd.
+   ///
+   /// The @a ndofs parameter can be used to indicate the total number of Dofs
+   /// associated with each component of @b vdim. If @a ndofs is -1 (the
+   /// default value), then the number of Dofs is <determined by the
+   /// FiniteElementSpace::GetNDofs().
    int DofToVDof(int dof, int vd, int ndofs = -1) const;
 
+   /// @brief Compute the inverse of the Dof to VDof mapping for a single
+   /// index @a vdof.
+   ///
+   /// @warning This method is only intended for use with positive indices.
+   /// Passing a negative value for @a vdof will produce an invalid result.
    int VDofToDof(int vdof) const
    { return (ordering == Ordering::byNODES) ? (vdof%ndofs) : (vdof/vdim); }
 
+   ///@}
+
+   /// @brief Remove the orientation information encoded into an array of dofs
+   /// Some basis function types have a relative orientation associated with
+   /// degrees of freedom shared between neighboring elements, see @ref ldof
+   /// for more information. An orientation mismatch is indicated in the dof
+   /// indices by a negative index value. This method replaces such negative
+   /// indices with the corresponding positive offsets.
+   ///
+   /// @note The name of this method reflects the fact that it is most often
+   /// applied to sets of @ref vdof "Vector Dofs" but it would work equally
+   /// well on sets of @ref ldof "Local Dofs".
    static void AdjustVDofs(Array<int> &vdofs);
 
-   /// Returns indexes of degrees of freedom in array dofs for i'th element.
+   /// @anchor getvdof @name Local Vector DoF Access Members
+   /// These member functions produce arrays of local vector degree of freedom
+   /// indices, see @ref ldof and @ref vdof. These indices can be used to
+   /// access entries in GridFunction, LinearForm, and BilinearForm objects
+   /// regardless of the value of @b vdim.
+   /// @{
+
+   /// @brief Returns indices of degrees of freedom for the @a i'th element.
+   /// The returned indices are offsets into an @ref ldof vector with @b vdim
+   /// not necessarily equal to 1. See also GetElementDofs().
```

Comment:
Should this also mention what the returned DofTransformation object is?

Reply:
It certainly wouldn't hurt. I'll add this.